### ISSUE TITLE
VanillaOS installer is stuck

### ISSUE THREAD

**Issue by mrunix00** [Vanilla-OS/vanilla-installer#418]
I have a somewhat unreliable 4G internet connection, and I have tried to install VanillaOS 2.0 3 times at this point and the same thing keeps happening, the installer gets stuck while downloading the necessary files for the system.
I'm expecting one of two solutions to this issue:
- Better error handling for the downloads (ex: Add a timeout for the downloads and when the timeout is exceeded restart the download for that part).
- Make the installer work offline.

**Comment by Vee-data-analytics** [comment]
yes I am also facing the exact same issue, I have installed it successfully before and had it working, itś always such a hassle to get it to install, now I am installing on a newly built PC and  I am on my 4th/5th attempt  and on my 3rd attempt It was nearly complete but didn´t  install grub
